src/apps/loyalty/server/index.tsx

```diff
@@ -72,53 +73,63 @@ app.get("/inquiries", (req, res) => {
     return res.redirect(req.baseUrl + "/login")
   }
 
-  let promise = IsomorphicRelay.prepareData({
-    Container: Inquiries,
-    queryConfig: new CurrentUserRoute(),
-  }, res.locals.networkLayer)
+  fetchCollectorProfile(req.user.get("accessToken"))
+    .then((info) => {
 
-  promise
-    .then(({data, props}) => {
-      const html = renderToString(<IsomorphicRelay.Renderer {...props} />)
-      const styles = styleSheet.rules().map(rule => rule.cssText).join("\n")
-      res.send(renderPage({
-        styles,
-        html,
-        entrypoint: "/bundles/inquiries.js",
-        bootstrapData: `var DATA = ${JSON.stringify(data)}; var USER_DATA = ${JSON.stringify(req.user.toJSON())};`,
-      }))
-    })
+      if (info.loyalty_applicant_at) {
+        return res.redirect(req.baseUrl + "/thank-you")
+      }
+
+      if (info.confirmed_buyer_at) {
+        markCollector(req.user.get("accessToken"))
```

Comment:
For some reason this is causing the dreaded `Error: Can't set headers after they are sent.` and a node server crash!